packages/nextjs/app/page.tsx

```diff
@@ -19,6 +27,57 @@ const Home: NextPage = () => {
             <span>To Be Implemented</span>
           </p>
         </div>
+        <div className="flex items-center flex-col flex-grow">
+          {isCheckedInEventsLoading ? (
+            <div className="flex justify-center items-center mt-10">
+              <span className="loading loading-spinner loading-lg"></span>
+            </div>
+          ) : (
+            <div className="mt-8">
+              <div className="text-center mb-4">
+                <span className="block text-2xl font-bold">Builders List</span>
+              </div>
+              <div className="overflow-x-auto shadow-lg">
+                <table className="table table-zebra w-full">
+                  <thead>
+                    <tr>
+                      <th className="bg-primary text-lg">Address </th>
+                      <th className="bg-primary text-lg">Info</th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    {!checkedInEvents || checkedInEvents.length === 0 ? (
+                      <tr>
+                        <td colSpan={2} className="text-center text-3xl">
+                          Loading builders...
+                        </td>
+                      </tr>
+                    ) : (
+                      checkedInEvents
+                        ?.slice()
```

Comment:
We don't need slice right ? lets remove it ? Also since we already already making `checkedInEvents` is not undefined above maybe remove `?.`  too

Reply:
I will make sure it is working with the changes you suggest and submit again

Reply:
Also, if I remove the `slice` the order is changed... it shows you at the top the last member that checked in. 

I don't understand 100% why it works with `slice` tbh... could be that since `slice` creates a shallow copy of the array, that shallow array is what actually is rendered?

Reply:
Ohh the thing is we are using `.reverse` which does an inplace sort and mutate the array itself. 

If we don't use `.slice` just do `checkedInEvents.reverse()` it doesn't work because since `checkedInEvents` is state variable and react will re-render component only when we update state via `setState` function other wise react wont render component. 

works with `.slice` since `.slice` first create new instance and `.reverse` update the new array instead of state directly. 

An alternate solution will to do use `checkedInEvents.toReversed()` directly instead of `.slice().reverse()`

Reply:
very interesting... thanks for the explanation!... for now i think i will stick with `.slice().reverse()` if you don't mind

Reply:
> for now i think i will stick with .slice().reverse() if you don't mind

Ohh yeah, I think it's fine but we should avoid using methods which mutate data inplace atleast while rendering things on UI